src/py21cmfast/src/Stochasticity.c

```diff
@@ -232,17 +237,20 @@ int set_prop_rng(gsl_rng *rng, int from_catalog, double *interp, float * input,
     //find log(property/variance) / mean
     double prop1 = gsl_ran_ugaussian(rng);
     double prop2 = gsl_ran_ugaussian(rng);
+    double prop3 = gsl_ran_ugaussian(rng);
 
     //Correlate properties by interpolating between the sampled and descendant gaussians
     //THIS ASSUMES THAT THE SELF-CORRELATION IS IN THE LOG PROPRETY, NOT THE PROPERTY ITSELF
     //IF IT SHOULD BE IN LINEAR SPACE, EXPONENTIATE THE RANDOM VARIABLES
     if(from_catalog){
         prop1 = (1-interp[0])*prop1 + interp[0]*input[0];
         prop2 = (1-interp[1])*prop1 + interp[1]*input[1];
+        prop3 = (1-interp[2])*prop1 + interp[2]*input[2];
```

Comment:
I don't see why this (and below) shouldn't just be in a loop over the inputs

Reply:
I'm actually planning on going the other direction with this, and changing `prop1`,`prop2`... to `rng_star`,`rng_sfr`...

Reply:
That sounds even better, thanks!